### PR TITLE
RDM-1923 Multiple roles endpoint

### PR DESCRIPTION
**Pull request by tzarsmango**
- Added a new endpoint for querying multiple user roles.
- Parametrized SSL usage with a new ENV param (defaults:false)
- Merged several ITs for faster testing cycle

### CODE REVIEW

File: application/src/test/java/uk/gov/hmcts/net/ccd/definition/store/rest/MultipleControllersEndpointIT.java [hunk 1]
```diff
@@ -111,4 +117,44 @@ public void shouldReturnSingleWizardPageForTestComplexAddressBookCase() throws E
                 hasItem(Matchers.hasProperty("caseFieldId", containsString("ContectEmail"))))))
         );
     }
+
+    // To be @Nested - UserRoleController
+    @Test
+    public void shouldReturnUserRolesForDefinedRoles() throws Exception {
+        givenUserProfileReturnsSuccess();
```
Comment (andrewfolga) [inline application/src/test/java/uk/gov/hmcts/net/ccd/definition/store/rest/MultipleControllersEndpointIT.java]: Why are we having this mocking here? Is it being used in the flow?

File: application/src/test/java/uk/gov/hmcts/net/ccd/definition/store/rest/MultipleControllersEndpointIT.java [hunk 2]
```diff
@@ -111,4 +117,44 @@ public void shouldReturnSingleWizardPageForTestComplexAddressBookCase() throws E
...
+            .andExpect(MockMvcResultMatchers.status().isOk())
+            .andReturn();
+        final List<UserRole> userRoles = mapper.readValue(result.getResponse().getContentAsString(), TypeFactory.defaultInstance().constructType(new TypeReference<List<UserRole>>(){}));
+        assertThat(userRoles, hasSize(6));
+    }
+
+    // To be @Nested - CaseDefinition Controller
+    @Test
+    public void shouldReturnJurisdictions() throws Exception {
+        givenUserProfileReturnsSuccess();
```
Comment (andrewfolga) [inline application/src/test/java/uk/gov/hmcts/net/ccd/definition/store/rest/MultipleControllersEndpointIT.java]: Same as above.

File: domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/UserRoleServiceImplTest.java [hunk 3]
```diff
@@ -109,15 +116,50 @@ public void shouldUpdate_whenSaveRole() {
...
+            () -> assertThat(userRoles.get(0).getRole(), is(roleNames[0])),
+            () -> assertThat(userRoles.get(0).getSecurityClassification(), is(RESTRICTED)),
+            () -> assertThat(userRoles.get(1).getId(), is(-3)),
+            () -> assertThat(userRoles.get(1).getRole(), is(roleNames[2])),
+            () -> assertThat(userRoles.get(1).getSecurityClassification(), is(PUBLIC))
+        );
+    }
+
+    @Test
+    @DisplayName("should return null userRoles if no roles defined")
```
Comment (andrewfolga) [inline domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/UserRoleServiceImplTest.java]: Could we fixed the display name here?

File: domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/UserRoleServiceImplTest.java [hunk 4]
```diff
@@ -10,29 +11,36 @@
...
+    @BeforeEach
+    void setUp() {
         repository = mock(UserRoleRepository.class);
         service = new UserRoleServiceImpl(repository);
     }
 
     @Test
-    public void getRole_shouldGetRole() {
+    @DisplayName("should return role if defined")
+    void getRole_shouldGetRole() {
```
Comment (vlaurin) [inline domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/UserRoleServiceImplTest.java]: I would suggest using `@Nested` here to group tests and remove the need for a `getRole` prefix.

File: rest-api/src/test/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/UserRoleControllerTest.java [hunk 5]
```diff
@@ -68,41 +70,87 @@ public void setUp() {
...
+            get(URI_TEMPLATE.expand(uriVariables)))
             .andExpect(status().isOk())
             .andReturn();
 
         final UserRole userRole = MAPPER.readValue(mvcResult.getResponse().getContentAsString(), UserRole.class);
-        assertThat(userRole.getId(), is(-6));
-        assertThat(userRole.getRole(), is(ROLE_DEFINED));
-        assertThat(userRole.getSecurityClassification(), is(mockUserRole.getSecurityClassification()));
+
+        assertAll(
```
Comment (andrewfolga) [inline rest-api/src/test/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/UserRoleControllerTest.java]: 👍

File: rest-api/src/test/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/UserRoleControllerTest.java [hunk 6]
```diff
@@ -68,41 +70,87 @@ public void setUp() {
     }
 
     @Test
-    public void shouldGetRole_whenRoleExists() throws Exception {
-
+    void shouldGetRole_whenRoleExists() throws Exception {
```
Comment (vlaurin) [inline rest-api/src/test/java/uk/gov/hmcts/ccd/definition/store/rest/endpoint/UserRoleControllerTest.java]: Same comment regarding the use of `@Nested`.

File: application/src/test/java/uk/gov/hmcts/net/ccd/definition/store/rest/MultipleControllersEndpointIT.java [hunk 7]
```diff
@@ -111,4 +117,44 @@ public void shouldReturnSingleWizardPageForTestComplexAddressBookCase() throws E
...
+
+        List<Jurisdiction> jurisdictions = mapper.readValue(result.getResponse().getContentAsString(),
+            TypeFactory.defaultInstance().constructType(new TypeReference<List<Jurisdiction>>(){}));
+
+        assertAll(
+            () -> assertThat(jurisdictions, hasSize(1)),
+            () -> assertThat(jurisdictions, hasItem(allOf(hasProperty("id", is("TEST")),
+                hasProperty("name", is("Test")),
+                hasProperty("name", is("Test"))
+            )))
```
Comment (adr1ancho) [inline application/src/test/java/uk/gov/hmcts/net/ccd/definition/store/rest/MultipleControllersEndpointIT.java]: 2 properties "name" ?

File: domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/UserRoleServiceImplTest.java (hunk 3 above)
Comment (tzarsmango) [reply domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/UserRoleServiceImplTest.java]: 👍

File: application/src/test/java/uk/gov/hmcts/net/ccd/definition/store/rest/MultipleControllersEndpointIT.java (hunk 1 above)
Comment (tzarsmango) [reply application/src/test/java/uk/gov/hmcts/net/ccd/definition/store/rest/MultipleControllersEndpointIT.java]: removed this one

File: application/src/test/java/uk/gov/hmcts/net/ccd/definition/store/rest/MultipleControllersEndpointIT.java (hunk 2 above)
Comment (tzarsmango) [reply application/src/test/java/uk/gov/hmcts/net/ccd/definition/store/rest/MultipleControllersEndpointIT.java]: This is required for the definition import phase

File: domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/UserRoleServiceImplTest.java (hunk 4 above)
Comment (tzarsmango) [reply domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/UserRoleServiceImplTest.java]: 👍

File: application/src/test/java/uk/gov/hmcts/net/ccd/definition/store/rest/MultipleControllersEndpointIT.java (hunk 7 above)
Comment (tzarsmango) [reply application/src/test/java/uk/gov/hmcts/net/ccd/definition/store/rest/MultipleControllersEndpointIT.java]: will remove duplicate

**Pull request merged**